### PR TITLE
Fix #14074 Mysql parser with XA statements

### LINKED ISSUE #14074: MySql parser has some grammar issues with XA statements
## Bug Report

Please refer to https://dev.mysql.com/doc/refman/5.7/en/xa-statements.html

```
XA {START|BEGIN} xid [JOIN|RESUME]

XA END xid [SUSPEND [FOR MIGRATE]]

XA PREPARE xid

XA COMMIT xid [ONE PHASE]

XA ROLLBACK xid

XA RECOVER [CONVERT XID]
```

### PR DESCRIPTION
**Pull request by zhfeng**
Fixes #14074 

Changes proposed in this pull request:
- Fix the grammer
- Add  ```op``` and ```xid``` attribute in ```XAStatement```
- Add a test

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/TCLStatement.g4 [hunk 1]
```diff
@@ -90,5 +90,17 @@ lockOption
...
+    : string_ | HEX_DIGIT_ | BIT_NUM_
+    ;
+
+bqual
+    : string_ | HEX_DIGIT_ | BIT_NUM_
+    ;
+
+formatID
+    : NUMBER_
+    ;
```
Comment (TeslaCN) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/TCLStatement.g4:106]: Please add a line break here.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/TCLStatement.g4 [hunk 2]
```diff
@@ -90,5 +90,17 @@ lockOption
     ;
 
 xid
-    : string_ (COMMA_ string_)* numberLiterals?
+    : gtrid ((COMMA_ bqual) (COMMA_ formatID)?)?
```
Comment (jingshanglu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/TCLStatement.g4:93]: @zhfeng Maybe it's better like this:
```
xid
    : gtrid=textString (COMMA_ bqual=textString (COMMA_ formatID=NUMBER_)?)?
    ;

```

and delete `gtrid`,`bqual`,`formatID`

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/tcl/XAStatement.java [hunk 3]
```diff
@@ -17,10 +17,20 @@
...
 package org.apache.shardingsphere.sql.parser.sql.common.statement.tcl;
 
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
 import org.apache.shardingsphere.sql.parser.sql.common.statement.AbstractSQLStatement;
 
 /**
  * Set transaction statement.
  */
```
Comment (jingshanglu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/tcl/XAStatement.java:27]: XA statement?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/TCLStatement.g4 (hunk 2 above)
Comment (zhfeng) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/TCLStatement.g4:93]: good catch ! I will check it. Thanks !

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/tcl/XAStatement.java (hunk 3 above)
Comment (zhfeng) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/tcl/XAStatement.java:27]: yeah, that's a typo.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/TCLStatement.g4 (hunk 2 above)
Comment (zhfeng) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/TCLStatement.g4]: Done

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/tcl/XAStatement.java (hunk 3 above)
Comment (zhfeng) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/tcl/XAStatement.java]: fixed

**Review by jingshanglu — APPROVED** [review]

**Pull request merged**